### ISSUE TITLE
Package versioning question

### ISSUE THREAD

**Issue by YoshinoriN** [hexojs/hexo#3630]
@hexojs/core

As you know now we are dropping Node.js v6 from all of the packages https://github.com/hexojs/hexo/issues/3508. We will publish new version near the feature. 

I have a question about release new version. Some packages not yet bump to v1 (major version). 

My understanding, drop Node.js v6 is relevant to breaking change.

But, Node.js(npm)'s versioning seems a little bit special. So, my question is "when bump v1"? 
What do you think & What should do?

---

Following packages are not yet bump to v1.

* https://github.com/hexojs/hexo-log
* https://github.com/hexojs/hexo-util
* https://github.com/hexojs/hexo-browsersync
* https://github.com/hexojs/hexo-generator-category
* https://github.com/hexojs/hexo-generator-archive
* https://github.com/hexojs/hexo-generator-tag
* https://github.com/hexojs/hexo-generator-index
* https://github.com/hexojs/hexo-i18n
* https://github.com/hexojs/hexo-front-matter
* https://github.com/hexojs/hexo-server
* https://github.com/hexojs/hexo-clean-css
* https://github.com/hexojs/hexo-html-minifier
* https://github.com/hexojs/hexo-uglify
* https://github.com/hexojs/hexo-notify
* https://github.com/hexojs/hexo-deployer-openshift
* https://github.com/hexojs/hexo-deployer-heroku
* https://github.com/hexojs/hexo-deployer-rsync
* https://github.com/hexojs/hexo-renderer-stylus
* https://github.com/hexojs/hexo-pagination
* https://github.com/hexojs/hexo-filter-auto-spacing
* https://github.com/hexojs/hexo-migrator-rss
* https://github.com/hexojs/hexo-renderer-jade
* https://github.com/hexojs/hexo-renderer-dot
* https://github.com/hexojs/hexo-renderer-haml
* https://github.com/hexojs/hexo-renderer-coffeescript
* https://www.npmjs.com/package/hexo-renderer-ejs
* https://www.npmjs.com/package/hexo-generator-alias

Thanks :)

**Comment by tomap** [comment]
I don't see the "special" part of the npm versioning. Breaking change = major+1. Easy, simple

**Comment by tcrowe** [comment]
I think only "minor" change is required until those modules are es6.

https://semver.org/spec/v2.0.0.html#summary

While node's API has changed our API has not changed unless it goes to es6 or the function and class signatures change. So, segayuu was correct bumping warehouse major version because it went to es6 class.

Are my arguments correct?

**Comment by curbengh** [comment]
Dropping node 6 is a breaking change; say package A is installed as `^1.2.8`, if node 6 is dropped in v1.3.0, node 6 users would encounter issue when running `npm i`.

But, in the case of `v0.x.0`, `^0.3.4` would only update `0.3.x`, so it's safe to drop node 6 in `^0.4.0`.

I think it depends, if a package still doesn't _feel_ v1, then minor version bump is fine. I'm being vague on v1 here, because everyone's definition is different.

TL;DR I'm ok with major and minor bump, as long the update doesn't suddenly break node 6 users.

**Comment by YoshinoriN** [comment]
@tomap Thanks :)
Yes, basically I agree you. But, I confuse because many of npm package not released v1.0.0 (e.g. It seems that [marked](https://github.com/markedjs/marked) update minor `0.x` version when include breaking change.)

@tcrowe Thanks :)
But, not migrate **ES6** matter. I refer to **drop Node.js v6** :)

@curbengh
So, should we decide this matter each packages, before publish new release?

**Comment by curbengh** [comment]
I think patch version bump most probably don't need, but for `v0.x.0, e.g. v0.2.0 -> v0.3.0` or `vx.0.0, e.g. v2.0.0 -> v3.0.0`, it's better to notify team.

**Comment by tomap** [comment]
I was going to publish hexo-renderer-marked, but it depends on hexo-util, which needs to be published first in v1.0.0
I will publish a hexo-util v1.0.0-rc1, and use it in hexo-renderer-marked, to see if any issue araise
Then, I'll publish v1.0.0, and use it in hexo-renderer-marked and publish a v2.0.0 of it

Is it ok for everyone?

**Comment by tomap** [comment]
https://www.npmjs.com/package/hexo-util/v/1.0.0-rc1 published

**Comment by tomap** [comment]
I just published https://www.npmjs.com/package/hexo-renderer-marked/v/2.0.0-rc2 using hexo-util v1.0.0-rc1
And used it here: https://github.com/tomap/hexo-theme-minidyne-demo/pull/2
Result: https://5d45475fc02e2300091c0f53--hexo-theme-minidyne-demo.netlify.com/2016/11/12/weixin-app/
Looks good

Will also try to integrate both to hexo, and run tests

**Comment by tomap** [comment]
https://github.com/hexojs/hexo/pull/3646 => a few failed tests, due to breaking changes or bugs?

**Comment by YoshinoriN** [comment]
> I was going to publish hexo-renderer-marked, but it depends on hexo-util, which needs to be published first in v1.0.0

> I will publish a hexo-util v1.0.0-rc1, and use it in hexo-renderer-marked, to see if any issue araise
Then, I'll publish v1.0.0, and use it in hexo-renderer-marked and publish a v2.0.0 of it

> Is it ok for everyone?

Basically ok. I have already [mentioned hexo-util](https://github.com/hexojs/hexo-util/pull/57#issuecomment-517919768). Please give us time before publish hexo-util v1.0.0

**Comment by YoshinoriN** [comment]
We already updated major version of many packages. I think this issue closable.

Thanks :)
